lib/OnboardingMessageProvider.jsm

```diff
@@ -87,22 +105,72 @@ const ONBOARDING_MESSAGES = () => [
   },
   {
     id: "TRAILHEAD_3",
-    template: "trailhead",
     targeting: "trailheadInterrupt == 'cards'",
-    trigger: { id: "firstRun" },
-    includeBundle: {
-      length: 3,
-      template: "onboarding",
-      trigger: { id: "showOnboarding" },
-    },
     utm_term: "trailhead-cards",
+    ...TRAILHEAD_ONBOARDING_TEMPLATE,
   },
   {
     id: "TRAILHEAD_4",
     template: "trailhead",
     targeting: "trailheadInterrupt == 'nofirstrun'",
     trigger: { id: "firstRun" },
   },
+  {
+    id: "TRAILHEAD_5",
+    targeting: "trailheadInterrupt == 'modal_control'",
+    utm_term: "trailhead-modal_control",
+    ...TRAILHEAD_ONBOARDING_TEMPLATE,
+    content: {
+      className: "joinCohort",
+      title: { string_id: "onboarding-welcome-body" },
+      benefits: ["products", "knowledge", "privacy"].map(id => ({
+        id,
+        title: { string_id: `onboarding-benefit-${id}-title` },
+        text: { string_id: `onboarding-benefit-${id}-text` },
+      })),
+      learn: {
+        text: { string_id: "onboarding-welcome-learn-more" },
+        url: "https://www.mozilla.org/firefox/accounts/",
+      },
+      form: {
+        title: { string_id: "onboarding-join-form-header" },
+        text: { string_id: "onboarding-join-form-body" },
+        email: { string_id: "onboarding-join-form-email" },
+        button: { string_id: "onboarding-join-form-continue" },
+      },
+      skipButton: { string_id: "onboarding-start-browsing-button-label" },
+    },
+  },
+  {
+    id: "TRAILHEAD_6",
+    targeting: "trailheadInterrupt == 'modal_variant_a'",
+    utm_term: "trailhead-modal_variant_a",
```

Comment:
Probably good to sanity check that utm_term can be any length (or at least this long). At least amplitude seems to have very long utm_terms, so this is probably okay.

Reply:
verified no length limit and string with letters, underscore and hyphen passes [regex check in place](https://github.com/mozilla/application-services/blob/master/docs/product-portal/accounts/metrics.md)